### ISSUE TITLE
Clarify zfsonlinux "Roadmap" and/or future transition to the new github repository.

### ISSUE THREAD

**Issue by cloudchin** [zfsonlinux/zfs#9619]
#8937 # Describe the problem you're observing
The "Roadmap" link on the zfsonlinux main site and the github wiki point to the Github "Milestone" section with 0 open issues, indicating to curious but uninitiated users or developers that there is no plan for this ZFS project. This of course, is pretty far from the truth.

### Describe how to reproduce the problem
Most simply, go to https://github.com/zfsonlinux/zfs/milestones , which is used as the link for the "Roadmap" on the Wiki and on the https://zfsonlinux.org main site. More indirectly, start from those areas first looking for information regarding the future of the project and click on that link. ( https://github.com/zfsonlinux/zfs/wiki/Project-and-Community or https://zfsonlinux.org )

It's fairly jarring to see no plan of record ("We couldn’t find anything!"). Although curious users can find references to future versions 0.9 and 1.0 under closed milestones, it is a very poor medium for communicating the vision and plan for such a wonderful project.

I do realize this maybe a transitional moment due to what I found on the mailing archive: plans to move to a new cross-platform repo, but the provided link to that repo in the post returns "not found", and in any case, that is a substantial change to the future discoverability of the project and it's scope which, if anything, should be made crystal clear now more than ever.

Source:
https://openzfs.topicbox.com/groups/developer/T6aa3c033248cef9c/zol-repo-move-to-openzfs

Can we update these sections and formalize an announcement at the top of the README/Issues section (and probably on https://zfsonlinux.org and probably in the announcement mailing archive) about the project?

**Comment by Ornias1993** [comment]
As stated on the mailing list:

> will be moved

this (also) means: 

> Is not yet

So its obvious that there won't be anything on that url.
You didn't really get what "will be moved" meant, so what keeps more people from getting confused by putting it on the readme already? ;)

> it is a very poor medium for communicating the vision and plan for such a wonderful project.

I think thats pretty universally accepted and therefore not used anymore for ages now

If you want a roadmap, you got one right here:
> == More Info ==
> See the slides and video from my talk at the 2019 OpenZFS Developer Summit for more context and details.

But due to the nature of this project, there isn't one big central roadmap. There are multiple groups of people working on different subprojects (see the list of pullrequests). Byond "as soon as freebsd is integrated we will move to OpenZFS2.0", there is no real roadmap. 
Often there is one or a few "milestones" that warrant a new release, but it's not "set in stone" many-months ahead.

TL:DR
- Currently there is no change nor that is relevant for users when it comes to repo movement. 
- Some people are working on FreeBSD support, that's done when it's done.
- There isn't really a fixed/forced roadmap... multiple people work on multiple sub-projects

**Comment by Ornias1993** [comment]
I agree it's a bit of a "cluster fuck" however, maybe that video should be linked on more places. Because that video is pretty clear.
(you did watch it right?)

**Comment by ahrens** [comment]
I agree we should figure out if we're going to use github "milestones", and if not, update the project documentation to reflect that.

As @Ornias1993 mentioned, the OpenZFS / ZoL project relies on a variety of contributors who develop new features, and we have only a little influence on their schedules. In terms of project planning, the OpenZFS / ZoL project intends to provide releases on a semi-regular schedule (e.g. release X is coming in approximately Y quarters), but we intend to minimize the projects that each release is dependent on.  So it would be rare that we could say with 100% confidence that "Feature X which is not yet in master will be part of release Y."

**Comment by cloudchin** [comment]
> I think thats pretty universally accepted and therefore not used anymore for ages now

I actually was referring more specifically to the fact that they are "closed". Since none are open and the current version is .8x, I actually assumed those maybe still relevant. I don't have any issue with using the Milestones section at all. Actually I think it's a fairly simple way of "getting it done".

> If you want a roadmap, you got one right here:
Thanks, I may dive in soon over the holiday. However documented links labeled as "Roadmap" should then make this material accessible, if it's the current running best forward trajectory.

Alternatively, I think it would be better to say there is no clear consensus of forward direction which can solidify into a formal Roadmap, and then remove those links until a consensus is made and someone wishes to maintain it if that is the case. Or even just say we don't want to maintain it and remove those links.

> Because that video is pretty clear.
> (you did watch it right?)

No I did not, I completely missed it, thanks. However, I used the info in the mailing archive as an example of information that should be provided from "Roadmap" labelled links instead of a blank Milestones section, so I think my point still has merit.

In any case, I'm not super familiar with the project's inner workings but I do greatly appreciate the progress made towards getting ZFS out there. Thank you!

**Comment by grahamperrin** [comment]
Should https://github.com/zfsonlinux/zfsonlinux.github.com/issues be used for (other) site-related issues? It's the repo for 

> The Official ZFS on Linux Website http://zfsonlinux.org

**Comment by Ornias1993** [comment]
@grahamperrin You know, you are actually right...
If you make an issue there (some copy, pasta and a link would suffice), I think this one can/should be closed :)

**Comment by grahamperrin** [comment]
Before shifting and/or closing … maybe timely to update these two pages: 

* http://open-zfs.org/wiki/History
* http://open-zfs.org/wiki/Roadmap

– in the context of the most recent keynote. 

Cherry-picked, slides 13 and 15: 

![OpenZFS DevSummit keynote 2019 slide 13](https://user-images.githubusercontent.com/192271/72298243-89c91c00-3655-11ea-90d1-a4dc3a7ec726.png)

![OpenZFS DevSummit keynote 2019 slide 15](https://user-images.githubusercontent.com/192271/72298265-9483b100-3655-11ea-98f7-e175324e85cd.png)

**Comment by ikozhukhov** [comment]
need additional arrow from OpenZFS -> DilOS
because DilOS has additional patches/updates what are not available on illumos and has independent OpenZFS port.

**Comment by Ornias1993** [comment]
@ikozhukhov It's not meant to be covering all operating systems.
Just to describe the fact linux/bsd/openzfs is at the top of the tree... ;)

@grahamperrin agreed, but thats also not an issue with this repo.
Even worse, this repo should absolutely not be used (yet) to discuss the openzfs wiki.

**Comment by behlendorf** [comment]
Thanks for baring with us.  Over the last year we've taken some significant steps which I think will help us better communicate the plan going forward.   For example, we now have a common GitHub repository and documentation site for Linux and FreeBSD.  We've also newly enabled the [GitHub Discussion](https://github.com/openzfs/zfs/discussions) section which we can start using to using to top pin important information.   There's still a fair bit of documentation which needs to be updated but with help from the community we're making good progress. 

To address the specific issue mentioned here I've gone ahead and removed the "Roadmap" link from the  https://zfsonlinux.org website.  We may still choose to use milestones but as Matt mentioned we'd like to try and minimize blocking releases on specific features.  The site now also links to the OpenZFS documentation and repository.
